src/printer.rs

```diff
@@ -291,8 +291,13 @@ impl Printer {
         header_string
     }
 
-    pub fn print_seperator(&mut self) -> io::Result<()> {
-        self.buffer.print("\n")?;
+    // Each of the print_* functions adds an extra line separator at the end
+    // except for print_response_body. We are using this function when we have
+    // something to print after the response body.
+    pub fn print_separator(&mut self) -> io::Result<()> {
+        if self.print.request_body {
```

Comment:
I'm not sure I understand this. It looks at the `request_body` setting, but it comes right after the `response_body`, right?

Would it make more sense to add the newline into whichever of `print_request_body` and `print_response_body` needs it?

(I'm also ok with ignoring this altogether, I don't know if matching the exact number of newlines is important as long as there are no unreasonably short or long gaps.)

Reply:
>I'm not sure I understand this. It looks at the request_body setting, but it comes right after the response_body, right?

Aargh, I meant to check `self.print.response_body` instead of `self.print.request_body`.

>Would it make more sense to add the newline into whichever of print_request_body and print_response_body needs it?

`print_request_body` does already print a newline. In contrast, `print_response_body` doesn't add a newline because it is usually the last thing that would be printed (except if it is an intermediate response).

Reply:
Ahh, now everything makes sense.

It might be clearer to inline `print_separator` at this point but I think it all works out.